src/main/java/edit_user_public_profile_usecase/editUserPublicProfileInteractor.java

```diff
@@ -0,0 +1,77 @@
+package edit_user_public_profile_usecase;
+
+import Entities.User;
+import UserRegistrationUsecase.NewUserDSGateway;
+
+/**
+ * The edit user's public profile use case.
+ */
+public class editUserPublicProfileInteractor implements editUserPublicProfileInputBoundary {
+    final private NewUserDSGateway userDSGateway;
+    final private editUserPublicProfileOutputBoundary profileOutputBoundary;
+
+    public editUserPublicProfileInteractor(NewUserDSGateway userDSGateway,
+                                           editUserPublicProfileOutputBoundary profileOutputBoundary) {
+        this.userDSGateway = userDSGateway;
+        this.profileOutputBoundary = profileOutputBoundary;
+    }
+
+    /**
+     *
+     * @param requestModel the request model for the edit user's public profile use case.
+     * @return the new edited changes to the public profile.
+     */
+    @Override
+    public editUserPublicProfileResponseModel saveEdits(editUserPublicProfileRequestModel requestModel) {
+        /*Creating a failed response model*/
+        editUserPublicProfileResponseModel profileFailedResponseModel = new editUserPublicProfileResponseModel(
+                requestModel.getPreferences(),
+                requestModel.getCoursePreferences(),
+                requestModel.getBio(),
+                "");
+
+        /*Returning a failed view when preferences are left blank*/
+        for (String p: requestModel.getPreferences().values()) {
+            if (p.equals("")) {
+                profileFailedResponseModel.setMessage("Preferences was left blank.");
+                return profileOutputBoundary.prepareFailView(profileFailedResponseModel);
+            }
+        }
+
+        /*Returning a failed view when preferences are left blank*/
+        if (requestModel.getCoursePreferences().equals("")) {
+            profileFailedResponseModel.setMessage("Course preferences was left blank.");
+            return profileOutputBoundary.prepareFailView(profileFailedResponseModel);
+        }
+
+        /*Returning a failed view when profile bio is left blank*/
```

Comment:
I had a question about these "empty" cases. Are we sure that we don't want to let the user completely erase their preferences? Like for course preferences, if they decide not to fall under any "course" (doing a side project or something) should this actually cause an error? Because i'm not sure if i should implement a similar thing for my usecase.